server/src/utils.ts

```diff
@@ -67,7 +67,8 @@ export let findProjectRootOfFile = (
   if (foundRootFromProjectFiles != null) {
     return foundRootFromProjectFiles;
   } else {
-    const isDir = path.extname(source) === "";
+    const dirStat = fs.statSync(source);
+    const isDir = dirStat.isDirectory();
```

Comment:
IIRC, this is in a hot path where we shouldn't use `fs` functions (which is why this fn looks the way it looks now). So we need to figure out another way to do this, outside of the hot path. OTOH maybe it could be aggressively cached in some sane way (unlikely), or moved outside of the hot path entirely to some sort of setup function somewhere. Would need to be explored.